cmd/soroban-cli/src/token/wrap.rs

```diff
@@ -147,12 +147,14 @@ impl Cmd {
             ))
         })?;
 
-        snapshot::commit(state.1, ledger_info, &storage.map, &self.ledger_file).map_err(|e| {
-            Error::CannotCommitLedgerFile {
+        state.set_ledger_info(ledger_info);
```

Comment:
I feel like having separating `set_ledger_info` and `update_entries` methods is unfortunate when they are always set together in the CLI. Are there any use cases where this isn't the case? 

I'm fine with merging this either way, and addressing this in the sdk later if we think something needs to change.

Reply:
It felt natural to me because the two things are very separate in the host too. The two aren't bound together, but I can add a single method that does both and use that here. It might prevent footguns. I'll follow up with that.

Reply:
Maybe we should pass the entire host to the snapshot lib, and it could pull out the relevant info?

Reply:
Ledger snapshot is dependent on the host already anyway because it uses the snapshot source trait from the host. So I think it could do this. Added:
- https://github.com/stellar/rs-soroban-sdk/issues/799